src/test/kotlin/se/uulm/snowballr/backend/service/user/GetCurrentUserTest.kt

```diff
@@ -0,0 +1,41 @@
+package se.uulm.snowballr.backend.service.user
+
+import io.mockk.coEvery
+import io.mockk.every
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.TestSpecificException
+import se.uulm.snowballr.backend.auth.GrpcContext
+import se.uulm.snowballr.backend.service.MainServiceTest
+import java.util.UUID
+
+class GetCurrentUserTest : MainServiceTest() {
+    @Test
+    fun `When retrieving the user context fails, then an exception is thrown`() = runTest {
+        every { GrpcContext.getUserIdFromContext() } throws TestSpecificException()
+
+        assertThrows<TestSpecificException> { mainService.getCurrentUser() }
+    }
+
+    @Test
+    fun `When retrieving the user by id fails, then an exception is thrown`() = runTest {
+        val userId = UUID.randomUUID()
+        every { GrpcContext.getUserIdFromContext() } returns userId
+        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } throws TestSpecificException()
+
+        assertThrows<TestSpecificException> { mainService.getCurrentUser() }
+    }
+
+    @Test
+    fun `When retrieving the current user, then no exception is thrown`() = runTest {
+        val userId = UUID.randomUUID()
+        val user = DataBuilder.createExampleUser(id = userId)
+        every { GrpcContext.getUserIdFromContext() } returns userId
+        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } returns user
+
+        assertDoesNotThrow { mainService.getCurrentUser() }
+    }
```

Comment:
As I asked you in #250, why do you manually create the id?
You can just use the id of the user you created.

The creation of the id in the test before is fine, since we only need the id, but here, this makes no sense to me.

```suggestion
    fun `When retrieving the current user, then no exception is thrown`() = runTest {
        val user = DataBuilder.createExampleUser()
        every { GrpcContext.getUserIdFromContext() } returns user.id
        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } returns user

        assertDoesNotThrow { mainService.getCurrentUser() }
    }
```

Reply:
Sorry, I didn't have a look at #250 yet. But you are right. This doesn't make any sense at all. Good hint.